### ISSUE TITLE
Key error u'##'

### ISSUE THREAD

**Issue by tynopet** [textile/python-textile#46]
I try to transform this text:
```
# test
### test
## test
```
and get exception:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/dmitry/.local/lib/python2.7/site-packages/textile/core.py", line 1368, in textile
    return Textile(html_type=html_type).parse(text)
  File "/home/dmitry/.local/lib/python2.7/site-packages/textile/core.py", line 250, in parse
    text = self.block(text)
  File "/home/dmitry/.local/lib/python2.7/site-packages/textile/core.py", line 471, in block
    block = Block(self, tag, atts, ext, cite, line)
  File "/home/dmitry/.local/lib/python2.7/site-packages/textile/objects/block.py", line 32, in __init__
    self.process()
  File "/home/dmitry/.local/lib/python2.7/site-packages/textile/objects/block.py", line 124, in process
    self.content = self.textile.graf(self.content)
  File "/home/dmitry/.local/lib/python2.7/site-packages/textile/core.py", line 607, in graf
    text = self.textileLists(text)
  File "/home/dmitry/.local/lib/python2.7/site-packages/textile/core.py", line 297, in textileLists
    return pattern.sub(self.fTextileList, text)
  File "/home/dmitry/.local/lib/python2.7/site-packages/textile/core.py", line 349, in fTextileList
    self.olstarts[tl] = self.olstarts[tl] + 1
KeyError: u'##'
```

Can you fix this?

**Comment by ikirudennis** [comment]
Man, that packs a lot of ugly textile into a tiny test. I'm doing a massive rewrite of list handling, so I can't guarantee there will be a quick fix. I don't suppose there's any way to format that list correctly?

txstyle.org produces the following:
```html
<ol>
	<li>test
		<ol>
			<li>test</li>
		</ol></li>
	<ol>
		<li>test</li>
	</ol></li>
	</ol>
```
That seems pretty messy and not exactly what I'd expect given the input, but if you can't can't format the textile input better, is that what you're looking for? Can you provide an example of what you think the output should look like?

**Comment by tynopet** [comment]
Yes, this output is satisfactory. The main thing doesn't throw an error. Thank you.

**Comment by ikirudennis** [comment]
Hi, @tynopet. I think I've fixed both this issue and #47. Would you mind testing this to confirm it's fixed and not causing any other issues? `pip install git+https://github.com/textile/python-textile.git@develop#egg=textile` will have the fixes for both issues. Thanks, again.

**Comment by tynopet** [comment]
Thank you for fixes.
